### DISCUSSION TITLE
405 Allow header is removed when using a custom MethodNotAllowed handler

### DISCUSSION THREAD

**Discussion by BoilingSoup** [go-chi/chi#922] [Q&A]
By default, Chi sends an `Allow` response header with the allowed methods when handling a 405.
https://github.com/go-chi/chi/pull/776

When I use a custom 405 handler the `Allow` header is no longer present in responses.

```
r := chi.NewRouter()
r.MethodNotAllowed(notAllowedJSONHandler)

// "Allow" response header is not automatically set any more.

```

I'm not sure if this is intended behavior, so I'd like to discuss :)  

I think it would be a good default to always send the `Allow` header for 405 responses, even if using a custom handler.

**Comment by VojtechVitek** [comment]
I think this makes sense. Free to submit a PR.
